src/test/java/org/fundacionjala/trello/pages/card/CardPage.java

```diff
@@ -35,15 +48,41 @@ public boolean isDisplayed() {
         return true;
     }
 
-    public List<String> getAllCardNames(final String listName) {
-        String getCardsList = String.format(CARD_NAMES_LIST, listName);
-        action.waitForVisibilityOfAllElements(cardsList);
-        List<WebElement> getCards = driver.findElements(By.xpath(getCardsList));
+    public List<String> getCardNamesInList(final String listName) {
+        List<WebElement> getCards = getCardElementsInList(listName);
         List<String> list = new ArrayList<>();
         for (WebElement getCard : getCards) {
             String text = getCard.getText();
             list.add(text);
         }
         return list;
     }
+
+    public List<WebElement> getCardElementsInList(final String listName) {
+        String cardsInListLocator = String.format(GETS_ALL_CARDS_IN_A_LIST, listName);
+        action.waitForVisibilityOfAllElements(cardsList);
+        List<WebElement> getCards = driver.findElements(By.xpath(cardsInListLocator));
+        return getCards;
+    }
+
+    public CardUpdateForm navigateToCard(final String listName, final String cardName) {
+        List<WebElement> getCards = getCardElementsInList(listName);
+        for (WebElement getCard : getCards) {
+            if (getCard.getText().equals(cardName)) {
+                getCard.click();
+                return new CardUpdateForm(driver);
+            }
+        }
+        return null;
```

Comment:
It's not a good practice return null in a design method.
Move the line 73 to 76 in order to always return `new CardUpdateForm(driver);`

Reply:
ok, thank you